tests/grid_test.py

```diff
@@ -1,126 +1,357 @@
 import numpy
 import pytest
+import scipy.stats.qmc
 
-from smolyay.basis import (ChebyshevFirstKind, BasisFunctionSet, 
-                           NestedBasisFunctionSet)
-from smolyay.grid import (IndexGridGenerator, SmolyakGridGenerator,
-                          TensorGridGenerator, generate_compositions)
+import smolyay
 
 
-def test_error_basis_set_type():
-    """Test if raise TypeError works for IndexGridGenerator."""
-    basis_set = "not a BasisFunctionSet"
+@pytest.mark.parametrize(
+    "random_point_set",
+    [
+        smolyay.samples.UniformRandomPointSet,
+        smolyay.samples.LatinHypercubeRandomPointSet,
+        smolyay.samples.HaltonRandomPointSet,
+        smolyay.samples.SobolRandomPointSet,
+    ],
+    ids=["Uniform", "Latin", "Halton", "Sobol"],
+)
+def test_random_initalize(random_point_set):
+    """Test that the random point set initializes correctly"""
+    f = random_point_set([[-10, 10], [0, 2]], 64, 1234)
+    assert numpy.array_equal(f.domain, [[-10, 10], [0, 2]])
+    assert f.num_dimensions == 2
+    assert f.num_points == 64
+    assert isinstance(f.num_points, int)
+    assert f.seed == 1234
+    assert isinstance(f.seed, int)
+    f.domain = [-10, 10]
+    assert numpy.array_equal(f.domain, [[-10, 10]])
+    f.num_points = 128.0
+    assert f.num_points == 128
+    assert isinstance(f.num_points, int)
+    f.seed = 40.0
+    assert f.seed == 40
+    assert isinstance(f.seed, int)
+
+
+@pytest.mark.parametrize(
+    "qmc_point_set",
+    [
+        smolyay.samples.LatinHypercubeRandomPointSet,
+        smolyay.samples.HaltonRandomPointSet,
+        smolyay.samples.SobolRandomPointSet,
+    ],
+    ids=["Latin", "Halton", "Sobol"],
+)
+def test_random_qmc_initalize(qmc_point_set):
+    """Test that the random point set initializes correctly"""
+    f = qmc_point_set([[-10, 20]], 64, 5678, True, "random-cd")
+    assert numpy.array_equal(f.domain, [[-10, 20]])
+    assert f.num_dimensions == 1
+    assert f.num_points == 64
+    assert f.seed == 5678
+    assert isinstance(f.scramble, bool)
+    assert f.scramble == True
+    assert f.optimization == "random-cd"
+    f.scramble = 0
+    assert isinstance(f.scramble, bool)
+    assert f.scramble == False
```

Comment:
I think Python prefers this:
```suggestion
    assert f.scramble is False
```
otherwise, use `assert not f.scramble`